### PR TITLE
Mozilla random id fix and test

### PR DESCRIPTION
**Pull request by eriktews**
The current implementation uses the _uuid_ module to generate random integers. As a result, the current ids are negative with a chance of 50%, which is probably not intended.

This PR changes the generation of the random ids to the _random.getrandbits_ function, which only returns positive values. Also, test is added that fails should one of the visit_id integers be out of range.

### CODE REVIEW

**Comment by eriktews** [comment]
I'm sorry but I think the build failure is not my fault. That looks like an unrelated problem.

**Comment by motin** [comment]
> I'm sorry but I think the build failure is not my fault. That looks like an unrelated problem.
Indeed it was. I restarted the Travis build and on the second attempt it worked...

**Comment by motin** [comment]
R+

File: automation/DataAggregator/S3Aggregator.py [hunk 1]
```diff
@@ -327,22 +327,22 @@ def save_configuration(self, openwpm_version, browser_version):
...
-        """Generate visit id as randomly generated 53bit UUIDs.
+        """Generate visit id as randomly generated positive integer less than 2^53.
 
         Parquet can support integers up to 64 bits, but Javascript can only
         represent integers up to 53 bits:
         https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER
         Thus, we cap these values at 53 bits.
         """
-        return (uuid.uuid4().int & (1 << 53) - 1) - 2**52
+        return random.getrandbits(53)
```
[the remark this answers is not in the document]
Comment (eriktews) [reply automation/DataAggregator/S3Aggregator.py]: As you need 54 bits to represent the number 2^53, this cannot happen.

File: automation/DataAggregator/S3Aggregator.py [hunk 2]
```diff
@@ -286,7 +286,7 @@ def __init__(self, manager_params, browser_params):
         self.dir = manager_params['s3_directory']
         self.bucket = manager_params['s3_bucket']
         self.s3 = boto3.client('s3')
-        self._instance_id = (uuid.uuid4().int & (1 << 32) - 1) - 2**31
+        self._instance_id = random.getrandbits(31)
```
[the remark this answers is not in the document]
Comment (eriktews) [reply automation/DataAggregator/S3Aggregator.py]: That might work, but then you should expect that roughly after 2^16 executions, it's likely that the same id appears twice or more often.

[the remark this answers is not in the document]
Comment (englehardt) [reply automation/DataAggregator/S3Aggregator.py]: I've made this change in #542.

[the remark this answers is not in the document]
Comment (englehardt) [reply automation/DataAggregator/S3Aggregator.py]: @eriktews This is an improvement over generating a 31 bit number within the positive half of a signed 32 bit integer. I'm going to close this in favor of #542
